plugins/chat/assets/javascripts/discourse/components/chat-user-info.gjs

```diff
@@ -1,10 +1,17 @@
 import Component from "@glimmer/component";
```

Comment:
You can implement all of this with a modifier:

```javascript
trackStatus = modifier((element, [user]) => {
  user.statusManager.startTrackingStatus();
  
  return () => {
    user.statusManager.stopTrackingStatus();
  }
})

<template>
  ...
  <UserStatusMessage {{this.trackStatus @user}} />
  ...
</template>
```

Might have made few mistakes in the syntax, and you might also have to add `...attributes` in the template of `UserStatusMessage`.

Let me know if it's not clear enough.

Reply:
It's a bit tricky. I can switch to use a modifier easily enough, but the issue is that it will only allow updating/deleting of existing user statuses. For those that don't have a status already then they won't live update when they add one (with the modifier approach).

One solution is to add a wrapper element and attach the modifier there but it will mean an extra empty element for those users that don't have a status yet. This is why I have the constructor approach to track the status on each user regardless of current status.